tools/cldr-code/src/main/java/org/unicode/cldr/util/SupplementalCalendarData.java

```diff
@@ -0,0 +1,269 @@
+package org.unicode.cldr.util;
+
+import com.google.common.base.Suppliers;
+import com.google.common.collect.ComparisonChain;
+import com.google.common.collect.ImmutableMap;
+import com.ibm.icu.util.Freezable;
+import com.ibm.icu.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import org.unicode.cldr.icu.LDMLConstants;
+
+public class SupplementalCalendarData implements Iterable<String> {
+    /** an <era> element */
```

Comment:
```suggestion
    /** an <era> element, which represents the era information from a calendar such as:
    * <pre>         
    * <calendarData><supplementalData>
    *    <calendar type="islamic">
    *      <calendarSystem type="lunar" />
    *      <eras>
    *          <era type="0" start="622-7-15" code="islamic" aliases="ah"/>
    *           <era type="1" end="622-7-14" code="islamic-inverse" aliases="bh"/>
    *      </eras>
    *   </calendar>
    * </pre>
    */
```

Reply:
or just link to the spec?

Reply:
The behavior of start/end is documented in https://www.unicode.org/reports/tr35/tr35-69/tr35-dates.html#Calendar_Data

(It could use some wordsmithing to clarify that "open-ended" applies when there isn't a series; probably start that first. I'll file a ticket for that.)

Reply:
Ticket: https://unicode-org.atlassian.net/browse/CLDR-18576